src/libstd/keyword_docs.rs

```diff
@@ -1167,12 +1185,31 @@ mod await_keyword {}
 
 #[doc(keyword = "dyn")]
 //
-/// Name the type of a [trait object].
+/// `dyn` is a prefix of a [trait object]'s type.
 ///
-/// The documentation for this keyword is [not yet complete]. Pull requests welcome!
+/// The `dyn` keyword is used to highlight that calling methods on a trait is to be achieved by
```

Comment:
```suggestion
/// The `dyn` keyword highlights that calls to methods on the associated `Trait` are dynamically dispatched.
```
(and nix rest of sentence)